### ISSUE TITLE
All input fields are masked even when mask setting is 'Relaxed'

### ISSUE THREAD

**Issue by asharafshahi** [microsoft/clarity#309]
We have setup a React SPA with Clarity masking set to Relaxed as we want to be able to see every data input entered by users.  However, the recordings always mask all input fields during user entry.  We also tried specifically adding data-clarity-unmask="true" to the `body` tag and also individual `input` tags to no avail.  Are there any other options to force Clarity to avoid masking altogether?

Thank you

**Comment by ClaritySupport** [comment]
@asharafshahi ,although the masking mode is relaxed, having an attribute “type” with value “tel” still makes the element sensitive. There are two options to unmask ,either to add “data-clarity-unmask” as attribute on the masked element or unmask the element manually through the project dashboard
check this link for more details: [https://learn.microsoft.com/en-us/clarity/clarity-masking](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fclarity%2Fclarity-masking&data=05%7C01%7Cv-vikasd%40microsoft.com%7Cefacfcc4c6ce41fe51cd08daaacdb602%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638010097834059506%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=BEmgOkpV4kyuaE0vxmmg02muJbmJgWQklt67%2BpC5RZs%3D&reserved=0)

**Comment by moneyaddixt** [comment]
one of the boxes that I want to unmask contains a "text" attribute, and when I try to unmask it either manually or through adding the "data-clarity-unmask" attribute it is still masked and not visible when I check the recordings feature on Microsoft clarity

**Comment by ClaritySupport** [comment]
Hello,

Could you please share below details to investigate further.


  1.  element ID or class.
  2.  Link to the page having this box/element.
  3.  Clarity Project ID.
  4.  Recordings having the box masked.



please share the recording having the issue using "Share option" as shown below to investigate the issue further.



  1.  [Graphical user interface, text, application  Description automatically generated]

Thanks,
Clarity Support Team
Visit our docs!: https://docs.microsoft.com/en-us/clarity<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.microsoft.com%2Fen-us%2Fclarity&data=05%7C01%7Cv-vidasa%40microsoft.com%7C343fe08f40a94e184c6408da28fedd21%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637867372409365494%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Q1zklwmipOGmVmg01HjPFwTVAwft75whSRoB8fFF%2BoY%3D&reserved=0>
***@***.******@***.***>
***@***.***

Confidentiality note: This e-mail, and any attachment to it, contains privileged and confidential information intended only for the use of the individual(s) or entity named in the e-mail.  If the reader of the e-mail is not the intended recipient, or the employee or agent responsible for delivering it to the intended recipient, you are hereby notified that reading it is strictly prohibited.  If you have received this e-mail in error, please immediately return it to the sender and delete it from your system.

**Comment by moneyaddixt** [comment]
below are the information required: 

**link to the page having this box** 
https://fadiaboalward.com/7digits_checkout/

**element ID or class.**
Name
id = billing_first_name
email box  :
id= billing_email

**clarity project ID:** 
eks6quxtop

**recording:**
https://clarity.microsoft.com/shared/recording/c7807112-c90a-4881-a8df-72af7346f84f

**Comment by ClaritySupport** [comment]
These are all input boxes. Which will be maksed by default.

***@***.***


Thanks,
Clarity Support Team
Visit our docs!: https://docs.microsoft.com/en-us/clarity<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.microsoft.com%2Fen-us%2Fclarity&data=05%7C01%7Cv-vidasa%40microsoft.com%7C343fe08f40a94e184c6408da28fedd21%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637867372409365494%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Q1zklwmipOGmVmg01HjPFwTVAwft75whSRoB8fFF%2BoY%3D&reserved=0>
***@***.******@***.***>
***@***.***

Confidentiality note: This e-mail, and any attachment to it, contains privileged and confidential information intended only for the use of the individual(s) or entity named in the e-mail.  If the reader of the e-mail is not the intended recipient, or the employee or agent responsible for delivering it to the intended recipient, you are hereby notified that reading it is strictly prohibited.  If you have received this e-mail in error, please immediately return it to the sender and delete it from your system.

**Comment by moneyaddixt** [comment]
Can I unmask them through adding data clarity unmask=true, or any other way ?

**Comment by ClaritySupport** [comment]
They cannot be unmasked and is by design, Masking content | Microsoft Learn<https://learn.microsoft.com/en-us/clarity/setup-and-installation/clarity-masking> for more details.

Thanks,
Clarity Support Team
Visit our docs!: https://docs.microsoft.com/en-us/clarity<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.microsoft.com%2Fen-us%2Fclarity&data=05%7C01%7Cv-vidasa%40microsoft.com%7C343fe08f40a94e184c6408da28fedd21%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637867372409365494%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Q1zklwmipOGmVmg01HjPFwTVAwft75whSRoB8fFF%2BoY%3D&reserved=0>
***@***.******@***.***>
***@***.***

Confidentiality note: This e-mail, and any attachment to it, contains privileged and confidential information intended only for the use of the individual(s) or entity named in the e-mail.  If the reader of the e-mail is not the intended recipient, or the employee or agent responsible for delivering it to the intended recipient, you are hereby notified that reading it is strictly prohibited.  If you have received this e-mail in error, please immediately return it to the sender and delete it from your system.

**Comment by midweste** [comment]
Agree with Vrq, we as developers watch these recordings to understand why we have particular issues.  Need a way to unmask anything

**Comment by prasadnaikpn** [comment]
Agreed this does not make sense and th issue si there since the beginning. There seems to be no way to unmask the same the input fields

**Comment by ClaritySupport** [comment]
Hi,


Thanks for using Clarity. Clarity masks sensitive content on your website by default. Clarity doesn't capture this masked content. For more details Please refer Masking content | Microsoft Learn<https://learn.microsoft.com/en-us/clarity/setup-and-installation/clarity-masking>

Content in the input boxes is masked in all modes<https://learn.microsoft.com/en-us/clarity/setup-and-installation/clarity-masking#masking-modes> and can't be customized. The input box includes the following but not limited to:

Account ID

Address

Card number

CCV

Code

Contact number including cell number, mobile number, telephone number

CVV

Date of Birth

Email address

Hidden content

Name

Password

Secret key

Social ID

SSN

Zip code




Thanks,

Clarity Support Team

Visit our docs!: https://docs.microsoft.com/en-us/clarity

***@***.******@***.***>

[cid:cc40a03d-7082-44ea-9401-9e0c8151d617]



Confidentiality note: This e-mail, and any attachment to it, contains privileged and confidential information intended only for the use of the individual(s) or entity named in the e-mail.  If the reader of the e-mail is not the intended recipient, or the employee or agent responsible for delivering it to the intended recipient, you are hereby notified that reading it is strictly prohibited.  If you have received this e-mail in error, please immediately return it to the sender and delete it from your system.



________________________________
